docs/bdd-introduction.md

```diff
@@ -0,0 +1,17 @@
+
+## Behavior Driven Development(BDD):
+
+   Behavior driven development (or BDD) is an software development technique that encourages collaboration between developers, QA and non-technical or business participants in a software project.
+
+   BDD focuses on obtaining a clear understanding of desired software behavior through discussion with stakeholders. It extends [TDD](docs/tdd-introduction) by writing test cases in a natural language that non-programmers can read. Behavior-driven developers use their native language in combination with the ubiquitous language of domain driven design to describe the purpose and benefit of their code. This allow the developers to focus on why the code should be created, rather than the technical details, and minimizes translation between the technical language in which the code is written and the domain language spoken by the business, users, stakeholders, project management, etc.
+     
+   Coming to reality, Customer is a person who is an expert in dealing the things other than Technology. They need not to be aware of Edge case scenarios,beta testing and Alpha testing and so on. Even if he(customer) is familiar of the Technology stack,the only thing that is evolving in much faster pace is business, which change at constant pace.
```

Comment:
> Coming to reality, 

This part is a bit weird. not sure what to do with it.

## suggestion:

A _Customer_ is a stakeholder who is an expert at dealing directly with the domain, rather than the technology responsible for presenting the domain. The customer need not be aware of edge cases, the difference between alpha and beta testing, unit and feature tests, etc. Even if the customer is familiar with the evolving development process of the technology stack behind the user interface, he or she is typically more concerned with the domain and the business - which themselves are often changing at a rapid pace.

[ even with this suggested edit, the paragraph is still a bit strange ]